apps/web/test/lib/getSchedule/expects.ts

```diff
@@ -0,0 +1,105 @@
+import { diff } from "jest-diff";
+import { expect } from "vitest";
+
+import type { Slot } from "@calcom/trpc/server/routers/viewer/slots/types";
+
+export const expectedSlotsForSchedule = {
+  IstWorkHours: {
+    interval: {
+      "1hr": {
+        allPossibleSlotsStartingAt430: [
+          "04:30:00.000Z",
+          "05:30:00.000Z",
+          "06:30:00.000Z",
+          "07:30:00.000Z",
+          "08:30:00.000Z",
+          "09:30:00.000Z",
+          "10:30:00.000Z",
+          "11:30:00.000Z",
+        ],
+        allPossibleSlotsStartingAt4: [
+          "04:00:00.000Z",
+          "05:00:00.000Z",
+          "06:00:00.000Z",
+          "07:00:00.000Z",
+          "08:00:00.000Z",
+          "09:00:00.000Z",
+          "10:00:00.000Z",
+          "11:00:00.000Z",
+        ],
+      },
+    },
+  },
+};
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace jest {
+    interface Matchers<R> {
+      toHaveTimeSlots(expectedSlots: string[], date: { dateString: string; doExactMatch?: boolean }): R;
+      toHaveNoTimeSlots(date: { dateString: string }): R;
+    }
+  }
+}
+
+expect.extend({
+  toHaveTimeSlots(
+    schedule: { slots: Record<string, Slot[]> },
+    expectedSlots: string[],
+    { dateString, doExactMatch }: { dateString: string; doExactMatch: boolean }
```

Comment:
Added `doExactMatch` to do strict testing. Ideally we should have doExactMatch to be always true but it fails other tests, so it is a separate task.